frontend/device/generic/device.lua

```diff
@@ -116,8 +117,14 @@ end
 -- ONLY used for Kobo and PocketBook devices
 function Device:onPowerEvent(ev)
     local Screensaver = require("ui/screensaver")
+    local network_manager = require("ui/network/manager")
     if (ev == "Power" or ev == "Suspend") and not self.screen_saver_mode then
         self.powerd:beforeSuspend()
+        self.wifi_was_on = network_manager:getWifiStatus()
```

Comment:
Not the last known one. I am just suggesting we always restore the network on/off state to the one set by the user (not detected before suspend). This is also the behavior for other systems like Android, KDE, OSX etc. We just have to turn networkmanager on and let it pick whatever network is available.